### PR TITLE
feat: add Faiss as VectorStore provider 

### PR DESCRIPTION
**Pull request by ewfian**
Add support for Faiss and also implemented a `loadFromPython` method, which can read VectorStore saved from [Python's implementation](https://python.langchain.com/en/latest/modules/indexes/vectorstores/examples/faiss.html#saving-and-loading).

TODO:
- [X] Improve the pickle parser and fully implement [Pickle protocol version 4](https://peps.python.org/pep-3154/). Extract pickle.js as an NPM package and then import it. The package will be here [pickleparser](https://github.com/ewfian/pickleparser).
- [x] Port `id` to `uuid` mapping from Python implementation.
- [x] Save mapping into JSON file.
- [x] Add examples.
   - [x] Create a new index from texts.
   - [x] Create a new index from a loader.
   - [x] Save an index to file and load it again.
   - [x] Load the saved file from Python.
- [x] Add test cases.
   - [x] Unit Test: `fromTexts` + `addVectors`.
   - [x] Unit Test: `fromDocuments` + `addVectors`.
   - [x] Unit Test: exceptions.
   - [x] Integration Test: save and load.
   - [x] Integration Test: Load from Python.
   - [x] Integration Test: Add documents after loaded from Python.
   - [x] Integration Test: Save and re-load after loaded from Python.
- [x] Add documentation.
  - [x] Create pages.
  - [x] Add basic information.
  - [x] Add example for how to read from Python.

### CODE REVIEW

File: examples/src/indexes/vector_stores/faiss_loadfrompython.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { FaissStore } from "langchain/vectorstores/faiss";
+import { OpenAIEmbeddings } from "langchain/embeddings/openai";
+
+export const run = async () => {
+  // Save the vector store to a directory
+  const directory = "faiss_index";
+
+  // Load the vector store from the same directory
+  const loadedVectorStore = await FaissStore.loadFormPython(
```
Comment (nfcampos) [inline examples/src/indexes/vector_stores/faiss_loadfrompython.ts:9]: typo

File: langchain/src/vectorstores/index.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@
 );
 
 export { HNSWLib } from "./hnswlib.js";
+export { FaissStore } from "./faiss.js";
```
Comment (nfcampos) [inline langchain/src/vectorstores/index.ts:7]: remove, this index file is deprecated

File: examples/src/indexes/vector_stores/faiss_loadfrompython.ts (hunk 1 above)
Comment (ewfian) [reply examples/src/indexes/vector_stores/faiss_loadfrompython.ts]: fixed

File: langchain/src/vectorstores/index.ts (hunk 2 above)
Comment (ewfian) [reply langchain/src/vectorstores/index.ts]: fixed

**Comment by nfcampos** [comment]
Thanks a lot for this @ewfian this is a great PR, sorry for the delay in getting this in

**Pull request merged**